app/main.py

```diff
@@ -1,13 +1,28 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"}
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+
+    return {
+        "errors": [format_linter_error(error) for error in errors],
+        "path": f"{file_path}",
+        "status": "failed"}
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+
+    return [{"errors": [],
+             "path": f"{key}",
+             "status": "passed"}
+            if value == [] else {
```

Comment:
look, you have the same "path" in both cases. also, if you have no errors - the `format_single_linter_file` function will return `[]` so you can create `errors` and `path` keys without the `if else` statement. Use `if else` only for `status` key